src/common/procedure/src/local.rs

```diff
@@ -368,38 +369,70 @@ pub struct LocalManager {
     procedure_store: Arc<ProcedureStore>,
     max_retry_times: usize,
     retry_delay: Duration,
-    remove_outdated_meta_task: RepeatedTask<Error>,
+    state: Arc<Mutex<LocalManagerState>>,
+    running: Arc<AtomicBool>,
```

Comment:
question: Could we move `state` (or `remove_outdated_meta_task`) and `running` into `ManagerContext`? I remember that it is already shared by all runners and acts as the state of the manager.

If `remove_outdated_meta_task` needs `ManagerContext`, we can only move `running`.